diff --git a/patches/E4_LB.patch b/patches/E4_LB.patch
deleted file mode 100644
index 9fc3957..0000000
--- a/patches/E4_LB.patch
+++ /dev/null
@@ -1,13 +0,0 @@
-diff --git a/src/main/scala/nutcore/isa/RVI.scala b/src/main/scala/nutcore/isa/RVI.scala
-index bdd214b..7f2dd76 100644
---- a/src/main/scala/nutcore/isa/RVI.scala
-+++ b/src/main/scala/nutcore/isa/RVI.scala
-@@ -122,7 +122,7 @@ object RV32I_LSUInstr extends HasInstrType {
-   def SW      = BitPat("b???????_?????_?????_010_?????_0100011")
- 
-   val table = Array(
--    LB             -> List(InstrI, FuType.lsu, LSUOpType.lb ),
-+    LB             -> List(InstrI, FuType.lsu, LSUOpType.lbu),
-     LH             -> List(InstrI, FuType.lsu, LSUOpType.lh ),
-     LW             -> List(InstrI, FuType.lsu, LSUOpType.lw ),
-     LBU            -> List(InstrI, FuType.lsu, LSUOpType.lbu),
diff --git a/src/main/scala/nutcore/isa/RVI.scala b/src/main/scala/nutcore/isa/RVI.scala
index bdd214b..1917537 100644
--- a/src/main/scala/nutcore/isa/RVI.scala
+++ b/src/main/scala/nutcore/isa/RVI.scala
@@ -93,7 +93,7 @@ object RV32I_BRUInstr extends HasInstrType {
     BNE            -> List(InstrB, FuType.bru, ALUOpType.bne),
     BLT            -> List(InstrB, FuType.bru, ALUOpType.blt),
     BGE            -> List(InstrB, FuType.bru, ALUOpType.bge),
-    BLTU           -> List(InstrB, FuType.bru, ALUOpType.bltu),
+    BLTU           -> List(InstrB, FuType.bru, ALUOpType.blt),
     BGEU           -> List(InstrB, FuType.bru, ALUOpType.bgeu)
   )
 
@@ -129,7 +129,7 @@ object RV32I_LSUInstr extends HasInstrType {
     LHU            -> List(InstrI, FuType.lsu, LSUOpType.lhu),
     SB             -> List(InstrS, FuType.lsu, LSUOpType.sb ),
     SH             -> List(InstrS, FuType.lsu, LSUOpType.sh ),
-    SW             -> List(InstrS, FuType.lsu, LSUOpType.sw)
+    SW             -> List(InstrS, FuType.lsu, LSUOpType.sh)
   )
 }
 
