diff --git a/src/main/scala/nutcore/isa/RVI.scala b/src/main/scala/nutcore/isa/RVI.scala
index bdd214b..7f2dd76 100644
--- a/src/main/scala/nutcore/isa/RVI.scala
+++ b/src/main/scala/nutcore/isa/RVI.scala
@@ -122,7 +122,7 @@ object RV32I_LSUInstr extends HasInstrType {
   def SW      = BitPat("b???????_?????_?????_010_?????_0100011")
 
   val table = Array(
-    LB             -> List(InstrI, FuType.lsu, LSUOpType.lb ),
+    LB             -> List(InstrI, FuType.lsu, LSUOpType.lbu),
     LH             -> List(InstrI, FuType.lsu, LSUOpType.lh ),
     LW             -> List(InstrI, FuType.lsu, LSUOpType.lw ),
     LBU            -> List(InstrI, FuType.lsu, LSUOpType.lbu),
